mesh/mesh.hpp

```diff
@@ -1388,6 +1398,13 @@ class GeometricFactors
        - NQ = number of quadrature points per element, and
        - NE = number of elements in the mesh. */
    Vector detJ;
+
+   /// Element restricted nodes
+   /** This array uses a column-major layout with dimensions (ND x SDIM x NE) where
+       - ND   = number of degrees of freedom per element,
+       - SDIM = space dimension of the mesh = mesh.SpaceDimension(), and
+       - NE   = number of elements in the mesh. */
+   Vector Enodes;
```

Comment:
Hmm, I'm not sure we want this here.

May just upon explicit request -- otherwise just use internal temporary.

Reply:
For example, add something like `ENODES` in the `FactorFlags` enum.